### ISSUE TITLE
Dithering constant

### ISSUE THREAD

**Issue by Oktai15** [pytorch/audio#371]
Why do `torchaudio.compliance.kaldi.fbank` and `torchaudio.compliance.kaldi.spectrogram` have so large `dither` default parameter (=1.0)? It very often just noises full output.

It's common to use dither around 0, e.g 0.00001 in QuartzNet, Jasper -- near to SOTA ASR models (https://github.com/NVIDIA/NeMo/blob/master/examples/asr/configs/quartznet15x5.yaml).

I want to notice that even in torchaudio tutorial we have dither = 0.0: https://pytorch.org/tutorials/beginner/audio_preprocessing_tutorial.html.

Also look at this issue and how it was resolved: https://github.com/pytorch/audio/issues/157

**Comment by vincentqb** [comment]
> Why do `torchaudio.compliance.kaldi.fbank` and `torchaudio.compliance.kaldi.spectrogram` have so large `dither` default parameter (=1.0)? It very often just noises full output.

Can you provide an example of code with noisy output using default value?

> It's common to use dither around 0, e.g 0.00001 in QuartzNet, Jasper -- near to SOTA ASR models (https://github.com/NVIDIA/NeMo/blob/master/examples/asr/configs/quartznet15x5.yaml).

Can you provide Kaldi's or other software's default value?

**Comment by Oktai15** [comment]
About example: as I already mentioned, check this your issue https://github.com/pytorch/audio/issues/157

**Comment by vincentqb** [comment]
Thanks for pointing this out. Yes, I agree. Let's use `dither=0.` as default then, unless we have a stronger contender.

**Comment by vincentqb** [comment]
Addresses part of #263

**Comment by popcornell** [comment]
Second this, the default right now makes the whole `torchaudio.compliace.kaldi` features totally unusable out-of-the-box. 
I spent one hour looking at possible bugs on labels only to find out that basically my model was fed noise because of the dither default value.

**Comment by Oktai15** [comment]
@popcornell I know that feel bro (the same problem I had had and after that I created this issue)

**Comment by vincentqb** [comment]
@popcornell @Oktai15 -- We're looking at what would be a good value to use. What would you say would be a reasonable nonzero value? What values do other packages use, and that you like?

@cpuhrsch -- I would have used 0. by default but the implementation says explicitly to specify `energy_floor` in that case, see [here](https://github.com/pytorch/audio/pull/119/files#r386478033). Do you have more context? Note that all the test where done with `dither=0.`: see [here](https://github.com/pytorch/audio/pull/119/files#r386496266) for the original testing, and [here](https://github.com/pytorch/audio/tree/master/test/assets/kaldi) where the second parameter (dither) in the file names for testing are all 0., e.g. `spec-XXX-0-...`.

In the absence of more information, I'd suggest `dither=1e-5`.

```python
import torchaudio
import matplotlib
%matplotlib inline
import matplotlib.pyplot as plt

filename = "steam-train-whistle-daniel_simon.mp3"
s, sr = torchaudio.load(filename)

spec0 = torchaudio.transforms.Spectrogram()(s)[0]
plt.imshow(spec0.log2().numpy(), cmap='gray')
plt.show()

spec1 = torchaudio.compliance.kaldi.spectrogram(s, dither=0.)
plt.imshow(spec1.t().numpy(), cmap='gray')
plt.show()

spec2 = torchaudio.compliance.kaldi.spectrogram(s, dither=1e-5)
plt.imshow(spec2.t().numpy(), cmap='gray')
plt.show()

# Mean absolute percent difference 
print(2*((spec1 - spec2).abs()/(spec1.abs() + spec2.abs())).mean())
# We see an average absolute percentage difference of 2.5%.
```

**Comment by vincentqb** [comment]
Based on [this discussion](https://github.com/pytorch/audio/pull/119/files#r386545216), we'll simply set `dither` to 0 and `energy_floor` to 1 by default. This also seems to behave very closely to a small value of dither, see below.
```
import torchaudio
import matplotlib
%matplotlib inline
import matplotlib.pyplot as plt

filename = "/Users/vincentqb/audio/test/assets/steam-train-whistle-daniel_simon.mp3"
s, sr = torchaudio.load(filename)

spec0 = torchaudio.transforms.Spectrogram()(s)[0]
plt.imshow(spec0.log2().numpy(), cmap='gray')
plt.show()

spec1 = torchaudio.compliance.kaldi.spectrogram(s, dither=0., energy_floor=1.)
plt.imshow(spec1.t().numpy(), cmap='gray')
plt.show()

spec2 = torchaudio.compliance.kaldi.spectrogram(s, dither=1e-6)
plt.imshow(spec2.t().numpy(), cmap='gray')
plt.show()

# Mean absolute percent difference 
print(2*((spec1 - spec2).abs()/(spec1.abs() + spec2.abs())).mean())
# We see an average absolute percentage difference of 0.16%.
```

**Comment by csukuangfj** [comment]
> Why do torchaudio.compliance.kaldi.fbank and torchaudio.compliance.kaldi.spectrogram have so large dither default parameter (=1.0)

Kaldi uses `1` as the default dither value. It is fine for Kaldi because waveform in kaldi
has a range [-32768, 32767].  `1` is relatively small compared to the maximum value 32767.

However, in torchaudio, 

```python
torchaudio.load(filename)
```

returns a tensor with values in the range [-1, 1]. So if you still use the default value `1` from
Kaldi, you will distort the audio signal.
